src/view/components/customerIdWrapper.jsx

```diff
@@ -0,0 +1,186 @@
+                                component={Textfield}
+                                componentClassName="u-fieldLong"
+                                supportDataElement
+                              />
+                            </div>
+                          </div>
+                          <div className="u-gapTop">
+                            <WrappedField
+                              name={`customerIds.${index}.hash`}
+                              component={Checkbox}
+                              label="Convert ID to sha256 hash"
+                            />
+                          </div>
+                          <div className="u-gapTop">
+                            <FieldLabel
+                              labelFor="authenticatedStateField"
+                              label="Authenticated State"
+                            />
+                            <div>
+                              <WrappedField
+                                id="authenticatedStateField"
+                                name={`customerIds.${index}.authenticatedState`}
+                                component={Select}
+                                componentClassName="u-fieldLong"
+                                options={authenticatedStateOptions}
+                              />
+                            </div>
+                          </div>
+                          <div className="u-gapTop">
+                            <WrappedField
+                              name={`customerIds.${index}.primary`}
+                              component={Checkbox}
+                              label="Primary"
+                            />
+                          </div>
+                          <div className="u-gapTop u-alignRight">
+                            <ModalTrigger>
+                              <Button
+                                id="deleteButton"
+                                icon={<Delete />}
+                                variant="action"
+                              />
+                              <Dialog
+                                title="Delete Customer ID"
+                                onConfirm={() => {
+                                  arrayHelpers.remove(index);
+                                }}
+                                confirmLabel="Delete"
+                                cancelLabel="Cancel"
+                              >
+                                Would you like to proceed?
+                              </Dialog>
+                            </ModalTrigger>
+                          </div>
+                        </Well>
+                      ))}
+                  </div>
+                </React.Fragment>
+              );
+            }}
+          />
+        </React.Fragment>
+      );
+      break;
+    case "error":
+      returnValue = <ErrorMessage>{`${showStuff.message}`}</ErrorMessage>;
+      break;
+    default:
+      returnValue = null;
```

Comment:
We should probably show the user a loading indicator while we load namespaces. Again, this can happen later when you actually implement namespace loading.